astropy/convolution/convolve.py

```diff
@@ -290,14 +287,21 @@ def convolve(array, kernel, boundary='fill', fill_value=0.,
         raise NotImplementedError('convolve only supports 1, 2, and 3-dimensional '
                                   'arrays at this time')
 
-    # If normalization was not requested, we need to scale the array (since
-    # the kernel is effectively normalized within the cython functions)
-    if not normalize_kernel and not kernel_sums_to_zero:
-        result *= kernel_sum
+    # So far, normalization has only occured for nan_treatment == 'interpolate'
+    # beacuse this had to happen within the Cython extension so as to ignore
+    # any NaNs
+    if normalize_kernel:
+        if not nan_interpolate:
+            result /= kernel_sum
+    else:
+        if nan_interpolate:
+            result *= kernel_sum
```

Comment:
For nan interpolation, the normalization had to occur during the core convolution, there is no other way around this. If normalization was not wanted, undo it here.